payment-handler/interfaces.https.any.js

```diff
@@ -0,0 +1,20 @@
+// META: script=/resources/WebIDLParser.js
+// META: script=/resources/idlharness.js
+
+"use strict";
+
+if (self.importScripts) {
+    importScripts("/resources/testharness.js");
+    importScripts("/resources/WebIDLParser.js", "/resources/idlharness.js");
+}
+
+// https://w3c.github.io/payment-handler/
+
+promise_test(async() => {
```

Comment:
Sorry for being overly picky here, but this should really be: 

```JS
promise_test(async() => {
    const text = await fetch("/interfaces/payment-handler.idl")
        .then(response => response.text());
    const idlArray = new IdlArray();
    idlArray.add_idls(text);
    idlArray.test();
    done();
}, "Payment handler interfaces.");
```

Reply:
No worries. So that I etch this into my memory, please provide your motivation for this change.

Reply:
Done.

Reply:
Oh, rule is that the variable name  should reflect its content. So, if it's `text`, it should contain text and not a promise; otherwise it should be "textPromise". See also in the spec: we use the same naming pattern for `resposePromise` etc.

Also, generally `.then()` is considered  an anti-pattern. In simple cases like the above for .text() it's usually fine tho... but chaining them or nesting them is usually not  good. 

Hope that helps.

Reply:
Kewl... Do you have a good list of such patterns and anti-patterns for future reference?

Reply:
Jakes article is pretty good (but doesn't exactly include these rules... sorta implied) https://developers.google.com/web/fundamentals/getting-started/primers/async-functions